doc/examples/applications/plot_3d_structure_tensor.py

```diff
@@ -0,0 +1,160 @@
+# of values?
+
+n_plane, n_Y, n_X, n_chan = data.shape
+v_min, v_max = data[:, :, :, 1].min(), data[:, :, :, 1].max()
+print(f'range: ({v_min}, {v_max})')
+
+#####################################################################
+# Let us visualize the middle slice of our 3D image.
+
+px.imshow(
+    data[n_plane // 2, :, :, 1],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'}
+)
+
+#####################################################################
+# Let us pick a specific region, which shows relative X-Y isotropy. In
+# contrast, the gradient is quite different (and, for that matter, weak) along
+# Z.
+
+sample = data[5:13, 380:410, 370:400, 1]
+step = 3
+cols = sample.shape[0] // step + 1
+_, axes = plt.subplots(nrows=1, ncols=cols, figsize=(16, 8))
+
+for it, (ax, image) in enumerate(zip(axes.flatten(), sample[::step])):
+    ax.imshow(image, cmap='gray', vmin=v_min, vmax=v_max)
+    ax.set_title(f'Plane = {5 + it * step}')
+    ax.set_xticks([])
+    ax.set_yticks([])
+
+#####################################################################
+# To view the sample data in 3D, run the following code:
+#
+# .. code-block:: python
+#
+#     import plotly.graph_objects as go
+#
+#     (n_Z, n_Y, n_X) = sample.shape
+#     Z, Y, X = np.mgrid[:n_Z, :n_Y, :n_X]
+#
+#     fig = go.Figure(
+#         data=go.Volume(
+#             x=X.flatten(),
+#             y=Y.flatten(),
+#             z=Z.flatten(),
+#             value=sample.flatten(),
+#             opacity=0.5,
+#             slices_z=dict(show=True, locations=[4])
+#         )
+#     )
+#     fig.show()
+
+#####################################################################
+# Compute structure tensor
+# ========================
+# About the brightest region (i.e., at X ~ 22 and Y ~ 17), we can see
+# variations (and, hence, strong gradients) along either X or Y over a typical
+# length of 3.
+
+px.imshow(
+    sample[0, :, :],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'},
+    title='Interactive view of bottom slice of sample data.'
+)
+
+#####################################################################
+# Therefore, we choose a 'width' of 3 for the window function.
+
+sigma = 3
+A_elems = feature.structure_tensor(sample, sigma=sigma)
+
+#####################################################################
+# We can then compute the eigenvalues of the structure tensor.
+
+eigen = feature.structure_tensor_eigenvalues(A_elems)
+eigen.shape
+
+#####################################################################
+# Where is the largest eigenvalue?
+
+coords = np.unravel_index(eigen.argmax(), eigen.shape)
+assert coords[0] == 0  # by definition
+coords
+
+#####################################################################
+# We are looking at a local property. Let us consider a tiny neighbourhood
+# of this maximum in the X-Y plane.
+
+eigen[0, coords[1], 20:23, 12:14]
```

Comment:
shouldn't we use `coords` instead of these hardcoded numbers (20:23, 12:14) here?
```Python
ycoords = slice(coords[2] - 1, coords[2] + 2)
zcoords = slice(coords[3] - 1, coords[3] + 2)

eigen[0, coords[1], ycoords, zcoords]
```

Reply:
Oops! Yes, indeed, although in this case the order of coordinates is different. It goes like: `eigen[i, zcoords, ycoords, xcoords]`, with `i = 0, 1, 2`, is/are the `(i + 1)`th largest eigenvalue(s) at position/neighbourhood `(xcoords, ycoords, zcoords)`.

Thanks to your review, I've noticed something inconsistent in my narrative: I'm showing the bottom plane (z = 0) but then I'm describing the situation at z = 3, since that's where the overall largest eigenvalue is found. Fixing now!

Reply:
Done in 521d4d235e86ec64e709e586fc864850b45044fb.